DogScepterLib/Core/GMData.cs

```diff
@@ -4,109 +4,195 @@
+        public int Major = 1;
+
+        /// <summary>
+        /// The minor version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Minor = 0;
+
+        /// <summary>
+        /// The release version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Release = 0;
+
+        /// <summary>
+        /// The build version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Build = 0;
+
+        /// <summary>
+        /// Indicates the bytecode format of the GameMaker data file.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.
+        /// For example, this has a value of <c>17</c> from version 2.2.1 since at least 2022.3. </remarks>
+        public byte FormatID = 0;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns chunks to 16 bytes.
+        /// </summary>
+        /// <remarks>GameMaker versions after Game Maker: Studio 2.2.2 use this. On older versions,
+        /// chunks are 4 byte aligned instead.</remarks>
+        public bool AlignChunksTo16 = true;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns strings to 4 bytes.
+        /// </summary>
+        public bool AlignStringsTo4 = true;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns backgrounds to 8 bytes.
+        /// </summary>
+        public bool AlignBackgroundsTo8 = true;
+
+        /// <summary>
+        /// TODO: Whether the GameMaker data file uses Pre-Create events for rooms and objects.
+        /// </summary>
+        public bool RoomObjectPreCreate = false;
+
+        /// <summary>
+        /// Whether the GameMaker data file contains various variable counts whose purpose is currently unknown.
+        /// </summary>
+        public bool DifferentVarCounts = false;
+
+        /// <summary>
+        /// Whether the GameMaker data file uses option flags in the OPTN chunk.
+        /// </summary>
+        public bool OptionBitflag = true;
+
+        /// <summary>
+        /// Indicates whether this GameMaker data file was run from the IDE.
+        /// </summary>
+        public bool RunFromIDE = false;
+
+        /// <summary>
+        /// Whether the audio group is builtin to the GameMaker data file.
+        /// </summary>
+        //TODO: shouldn't this be a bool, considering it can only hold values 0 and 1 and that at a low level those two are the same?
```

Comment:
No... this is just a number. No need to make it any more complicated than that.

Reply:
Might've gotten the intent of the variable wrong then.
From what I understood, it stores whether the audio group is builtin to the data file (false for everything past gms2 or gms1.1354.161.1000, true for everything earlier), which is how I've written that documentation. And the "whether" indicates a boolean value.
Is it supposed to do something else instead?

Reply:
Pretty sure it has to do with the default audio group ID, or something like that. I'm not 100% sure how it works myself.